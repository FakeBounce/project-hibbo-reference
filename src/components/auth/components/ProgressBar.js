import React from 'react';
import { Animated } from 'react-native';
import { shape, instanceOf, string } from 'prop-types';
import appStyles from 'styles/appStyles';

import { SContainerProgress, AProgessBar, SProgressStep } from '../styles';

const ProgressBar = ({ styling, color }) => {
  return (
    <SContainerProgress>
      <SProgressStep />
      <AProgessBar color={color} style={styling} />
    </SContainerProgress>
  );
};

ProgressBar.defaultProps = {
  color: appStyles.colors.black,
};

ProgressBar.propTypes = {
  color: string,
  styling: shape({
    width: instanceOf(Animated.Value).isRequired,
  }).isRequired,
};

export default ProgressBar;
