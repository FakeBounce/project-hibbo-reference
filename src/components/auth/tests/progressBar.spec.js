import {
  Animated,
} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProgressBar from '../components/ProgressBar';

describe('ProgressBar', () => {
  it('should render the component', () => {
    const styling = {
      width: new Animated.Value(100),
    };
    const component = renderer.create(
      <ProgressBar
        styling={styling}
      />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
