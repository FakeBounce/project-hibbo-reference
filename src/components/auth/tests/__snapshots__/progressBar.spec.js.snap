// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render the component 1`] = `
<View
  style={undefined}
>
  <View
    collapsable={undefined}
    style={
      Object {
        "width": 100,
      }
    }
  />
</View>
`;
